src/EditorFeatures/Core/Remote/SolutionChecksumUpdater.cs

```diff
@@ -202,57 +197,68 @@ await client.TryInvokeAsync<IRemoteAssetSynchronizationService>(
             cancellationToken).ConfigureAwait(false);
     }
 
-    private async ValueTask SynchronizeTextChangesAsync(
-        ImmutableSegmentedList<(Document oldDocument, Document newDocument)> values,
-        CancellationToken cancellationToken)
+    private void DispatchSynchronizeTextChanges(
+        Document oldDocument,
+        Document newDocument)
     {
-        var client = await RemoteHostClient.TryGetClientAsync(_workspace, cancellationToken).ConfigureAwait(false);
-        if (client == null)
-            return;
-
-        // this pushes text changes to the remote side if it can. this is purely perf optimization. whether this
-        // pushing text change worked or not doesn't affect feature's functionality.
-        //
-        // this basically see whether it can cheaply find out text changes between 2 snapshots, if it can, it will
-        // send out that text changes to remote side.
-        //
-        // the remote side, once got the text change, will again see whether it can use that text change information
-        // without any high cost and create new snapshot from it.
-        //
-        // otherwise, it will do the normal behavior of getting full text from VS side. this optimization saves
-        // times we need to do full text synchronization for typing scenario.
-        using var _ = ArrayBuilder<(DocumentId id, Checksum textChecksum, ImmutableArray<TextChange> changes, Checksum newTextChecksum)>.GetInstance(out var builder);
-
-        foreach (var (oldDocument, newDocument) in values)
+        _ = _threadingContext.JoinableTaskFactory.RunAsync(async () =>
```

Comment:
we should def doc the design here.  make it clear why we've written it this way.

note: you can also use IAsyncToken here to track the work, allowing you to properly be able to unit test as well.

Reply:
The intent is to fire-and-forget, but only yield the current thread if an async operation necessitates it.

Is GetChangeRanges expensive? It doesn't look to be to me, but maybe I'm missing an override? If it is expensive, the two conditions that use it could probably be removed, as they are very minor and specialized optimizations.

Will add some doc here

Reply:
If that's the intent, then I don't think the RunAsync() is doing anything for us here, right?

Reply:
I had _thought_ GetChangeRanges could potentially end up in a text diff, but maybe not.